packages/main/src/ColorValue.ts

```diff
@@ -0,0 +1,171 @@
+import {
+	getRGBColor,
+	HSLToRGB,
+	HEXToRGB,
+	RGBToHSL,
+	RGBtoHEX,
+} from "@ui5/webcomponents-base/dist/util/ColorConversion.js";
+import type {
+	ColorHSL,
+	ColorRGB,
+} from "@ui5/webcomponents-base/dist/util/ColorConversion.js";
+
+/**
+ * @class
+ *
+ * ### Overview
+ *
+ * The `ui5-color` class represents the `value` used in `ui5-color-picker`
+ *
+ * A color can be represented using rgb, hsl, or hex values. A color also has an alpha value.
+ * @public
+ */
+class ColorValue {
+	_rgb: ColorRGB;
+	_hsl: ColorHSL;
+	_alpha: number;
+	_hex: string;
+	_valid: boolean;
+
+	constructor(color: string = "rgba(255,255,255,1)") {
+		this._rgb = getRGBColor(color);
+		this._hsl = RGBToHSL(this._rgb);
+		this._hex = RGBtoHEX(this._rgb);
+		this._alpha = 1;
+		this._valid = true;
+	}
+
+	get RGB() {
+		return this._rgb;
+	}
+
+	get HSL() {
+		return this._hsl;
+	}
+
+	get H() {
+		return this._hsl.h;
+	}
+
+	get S() {
+		return this._hsl.s;
+	}
+
+	get L() {
+		return this._hsl.l;
+	}
+
+	get R() {
+		return this._rgb.r;
+	}
+
+	get G() {
+		return this._rgb.g;
+	}
+
+	get B() {
+		return this._rgb.b;
+	}
+
+	get Alpha(): number {
+		return this._alpha;
+	}
+
+	get HEX() {
+		return this._hex;
+	}
+
+	set RGB(value: ColorRGB) {
+		this._rgb = value;
+		if (this._valid) {
```

Comment:
When did `this._valid` got updated to correspond to the newly given RGB?

Reply:
You're right, the RGB and HSL setters didn't have their own validation because most of the time when the setters are called is from the individual value setters. The setters now have their own validation as well.